docs/DeveloperGuide.md

```diff
@@ -205,6 +211,43 @@ Classes used by multiple components are in the `seedu.trackermon.commons` packag
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Find command feature
+
+#### What it does 
+
+Looks for a show in a list of shows and displays all the shows that match the user's input. If the user's input contains no prefixes, `find` will do a general search through all fields in the `Show` class.
+
+#### Implementation
+After entering the find command, the tokenizer in parser will map any prefixes in the user's input to Trackermon's prefix syntax.Then, the parser will do a check whether there are any prefixes in the input. If prefixes are specified, a `FindCommand` object will be created with predicates looking through the specified prefixes. Else, a general show predicate will be created by scanning through the name, status and tag fields of the `Show` class. `FindCommand` is a class that inherits the `Command` abstract class. `FindCommand` implements the `execute()` method from the `Command` abstract class where on execution, it will scan through the shows in the model's list of shows and check if any shows match the user's input. The model is then udpated with the filtered show list.
+
+Given below is an example usage scenario and the step-by-step flow of the find command.
+
+Step 1: The user launches Trackermon and enters `find n/Shingeki no Kyojin s/watching` to find a show.
+
+Step 2: The find command will check and see whether any shows contain the name `Shingeki no Kyojin` and the status `watching` using the `Model#updateFilteredShowList` method.
+
+Step 3:
+`Model#updateFilteredShowList` will be called and model will be udpated.
+
+<div markdown="block" class="alert alert-info">
+**:information_source: Note:**<br>
+`find Shingeki no Kyojin` will also work, however it will scan through the name, status and tag fields instead of the name field only
+</div>
+
+The following activity diagram summarizes what happens when a user executes a valid find command:
+
+<img src="images/FindShowDiagram.png">
+
+#### Design considerations:
+##### Aspect: How find executes
+- **Alternative 1 (current choice):** The `find` command checks for keywords after the prefix. If there are no prefixes, it will perform a general search using `ShowcontainsKeywordsPredicate` which scans through the name, status and tag fields in the `Show` class. Else, `ArgumentMultimap#arePrefixesPresent` will check and return a new predicate for each value after each prefix if it is present and then generate a keyword predicate that will match values in that field. A for loop is implemented for the name and tag predicates. This will allow an AND search within a list of keywords within these two parameters. The user does not need to type in the full word for any find searches as each predicate uses a fragmented search.
```

Comment:
`ShowcontainsKeywordsPredicate` -> `ShowContainsKeywordsPredicate`

`AND search` should be quoted right? -> `"AND" search`

Reply:
Yes, I missed that out. Nice spot!